activerecord/test/cases/batches_test.rb

```diff
@@ -164,6 +164,40 @@ def test_find_in_batches_should_ignore_the_order_default_scope
     assert_equal posts(:welcome).id, posts.first.id
   end
 
+  def test_find_in_batches_should_error_on_ignore_the_order
+    assert_raise(ArgumentError) do
+      PostWithDefaultScope.find_in_batches(error_on_ignore: true){}
+    end
+  end
+
+  def test_find_in_batches_should_not_error_if_config_overriden
+    # Set the config option which will be overriden
+    prev = ActiveRecord::Base.error_on_ignored_order_or_limit
+    ActiveRecord::Base.error_on_ignored_order_or_limit = true
+    assert_nothing_raised do
+      PostWithDefaultScope.find_in_batches(error_on_ignore: false){}
+    end
+    # Set back to default
```

Comment:
Add this to `ensure`. This way if the test fail the value will set to the default again.